### DISCUSSION TITLE
"Unable to connect to mysql server." despite mariadb running

### DISCUSSION THREAD

**Discussion by Noxyntious** [osuAkatsuki/bancho.py#160] [Q&A] [answered]
Arch Linux 5.10.82-1-lts
hi, i'm getting the Unable to connect to mysql server. error despite making sure that both mariadb and redis are running, the gulag database existing, the login working, i've taken all troubleshooting steps that i could think of but nothing worked. keep in mind i am a complete mysql/mariadb noob. i've attached images of my .env and gulag.conf with (hopefully) all the personal data cut out.
![image](https://user-images.githubusercontent.com/44816759/148081240-d944bd82-f90e-4983-8784-7527ac080c09.png)
![image](https://user-images.githubusercontent.com/44816759/148081397-cb086cdf-f761-4419-bdc5-1f1983cde7f8.png)

**Comment by tsunyoku** [comment, marked as answer]
is your mysql definitely running on port 3306? (unless you've edited your mysql config, this should be the case)
can you also verify that you are able to get to the mysql shell?

**Reply by Noxyntious** [reply]
mysql is running on port 3306 (i think because i have not changed the config, can only be accessed from localhost though), and the shell is accessible 
![image](https://user-images.githubusercontent.com/44816759/148083413-3e2a9498-c2cd-439b-ad6e-c0b8c394ea1f.png)

**Reply by tsunyoku** [reply]
alright, in this case it's an issue on your end.

make sure the password you have provided is correct, and that the database name you've given exists. if you are using root user on linux, there's a high chance that it's using passwordless-auth so you should create a new user for mysql in that case.

let me know if there's any other issues :)

**Reply by cmyui** [reply]
we may be able to point you more into the right direction if you could post any error(s) you're getting when you start gulag - not clear to me by "unable to connect to mysql server" whether you mean the connection itself is failing, password auth is failing, etc.

if it's password auth failing then yes, it looks like you're root so it's likely using a password-less login when you connect via the shell

**Reply by Noxyntious** [reply]
the thing is, there is no other error, i just see this in the shell
![image](https://user-images.githubusercontent.com/44816759/148084577-5a0de5d5-6044-4b64-9209-769b1bed4323.png)

**Reply by cmyui** [reply]
could you try commenting out [this line](https://github.com/cmyui/gulag/blob/master/main.py#L25) (add a `#` before it) and see if that resolves your issue? it may also give you a 'real' error we can use

**Reply by Noxyntious** [reply]
i think i realize my issue now, gulag can't log in to the "root" account. going to try making a new mysql account to see if it will fix anything.
![image](https://user-images.githubusercontent.com/44816759/148085258-d9f68d36-273b-4675-b315-287fc800e0da.png)

**Reply by tsunyoku** [reply]
thought so, thanks for letting us know :D

**Reply by Noxyntious** [reply]
after creating a new user, it seems to work fine now! ty for helping!
![image](https://user-images.githubusercontent.com/44816759/148085681-316d9dc1-f3de-4cc5-a0e6-d65f225b40d8.png)
